ngraph_bridge/ngraph_pipelined_tensors.cc

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ * Copyright 2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#include "ngraph_bridge/ngraph_pipelined_tensors.h"
+
+using namespace std;
+namespace ng = ngraph;
+
+namespace tensorflow {
+
+namespace ngraph_bridge {
+
+IndexLibrary::IndexLibrary(size_t depth) : m_depth(depth) {
+  for (size_t i = 0; i < depth; i++) {
+    m_free_depth_indexes.insert(i);
+  }
+}
+
+void IndexLibrary::return_index(size_t id) {
+  if (m_depth == 0) {
+    throw std::runtime_error(
+        "Depth=0, so no one should be calling return_index");
+  } else {
```

Comment:
do you need else? as the throw will exit

Reply:
yes, in this case i don't see a good reason to have the elses. Just a carry-over from talways trying to write paired-up if-elses.

Also I should catch the errors and return TF status in TF code